x-pack/plugins/cross_cluster_replication/public/app/components/auto_follow_pattern_form.js

```diff
@@ -58,36 +90,48 @@ export class AutoFollowPatternFormUI extends PureComponent {
     remoteClusters: PropTypes.array.isRequired,
   }
 
-  state = {
-    autoFollowPattern: this.props.autoFollowPattern
-      ? {
+  constructor(props) {
+    super(props);
+
+    const isNew = this.props.autoFollowPattern === undefined;
+
+    const autoFollowPattern = isNew
+      ? getEmptyAutoFollowPattern(this.props.remoteClusters)
+      : {
         ...this.props.autoFollowPattern,
         ...getPrefixSuffixFromFollowPattern(this.props.autoFollowPattern.followIndexPattern)
-      }
-      : getEmptyAutoFollowPattern(this.props.remoteClusters),
-    isNew: typeof this.props.autoFollowPattern === 'undefined'
+      };
+
+    this.state = {
+      autoFollowPattern,
+      fieldsErrors: validateAutoFollowPattern(autoFollowPattern),
+      areErrorsVisible: false,
+      isNew,
+    };
   }
 
   onFieldsChange = (fields) => {
-    this.setState(({ autoFollowPattern }) => ({
-      autoFollowPattern: {
-        ...autoFollowPattern,
-        ...fields
-      }
-    }));
+    const errors = validateAutoFollowPattern(fields);
+
+    this.setState(updateFormValues(fields));
+    this.setState(updateFormErrors(errors));
   }
 
   onClusterChange = (remoteCluster) => {
     this.onFieldsChange({ remoteCluster });
   };
 
   onCreateLeaderIndexPattern = (indexPattern) => {
-    this.setState(({ autoFollowPattern }) => ({
-      autoFollowPattern: {
-        ...autoFollowPattern,
-        leaderIndexPatterns: [...autoFollowPattern.leaderIndexPatterns, indexPattern]
-      }
-    }));
+    const error = validateLeaderIndexPattern(indexPattern);
+
+    if (error) {
+      this.setState(updateFormErrors({ leaderIndexPatterns: { ...error, alwaysVisible: true } }));
+
+      // Return false to explicitly reject the user's input.
+      return false;
+    }
+
+    this.onFieldsChange({ leaderIndexPatterns: indexPattern });
```

Comment:
If we defined `leaderIndexPatterns` as an array containing both the previous index patterns and the new one, then I think we can remove the related conditions in `updateFormValues`, e.g. on line 58.